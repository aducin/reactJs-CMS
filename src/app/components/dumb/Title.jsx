import React from 'react';

const title = ( props ) => {
	let curStyle = props.curStyle !== undefined ? props.curStyle : null;
	return (
		<div class="container">
		    <div class="col-xs-12" style={curStyle}>
			    <h3>{props.title}</h3>
			</div>
		</div>
	);
}

export default title;