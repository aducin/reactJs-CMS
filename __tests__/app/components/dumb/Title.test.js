import React from 'react';
import TestRenderer from 'react-test-renderer';

import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Title from '../../../../src/app/components/dumb/Title.jsx';

configure({ adapter: new Adapter() });

const text = 'Just a test';
const style = { backgroundColor: 'red'};

const component = (
  <Title
    title={text}
    curStyle={style}
  />
);

const title = TestRenderer.create(component);

describe('Title snapshot', () => {
  it('Title renders correctly', () => {
    const tree = title.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Tests of the title component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(component);
  });
  it('H3 tag renders correct text', () => {
    expect(wrapper.find('h3').text()).toContain("Just a test");
  });
  it('Div renders style correctly', () => {
    expect(wrapper.find('div').first().children().first().props().style).toMatchObject({"backgroundColor": "red"});
  });
  it('H3 tag renders updated text', () => {
    wrapper.setProps({ title: 'Another text' });
    expect(wrapper.find('h3').text()).toContain("Another text");
  });
});
