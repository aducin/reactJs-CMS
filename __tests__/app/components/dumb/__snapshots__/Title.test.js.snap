// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshot Title renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="col-xs-12"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <h3>
      Just a test
    </h3>
  </div>
</div>
`;
